src/pretix/plugins/sendmail/views.py

```diff
@@ -36,13 +39,35 @@ def form_valid(self, form):
         self.request.event.log_action('pretix.plugins.sendmail.sent', user=self.request.user, data=dict(
             form.cleaned_data))
 
+        tz = pytz.timezone(self.request.event.settings.timezone)
+
         failures = []
+        self.output = {}
         for o in orders:
-            try:
-                mail(o.email, form.cleaned_data['subject'], form.cleaned_data['message'],
-                     None, self.request.event, locale=o.locale, order=o)
-            except SendMailException:
-                failures.append(o.email)
+            if self.request.POST.get("action") == "preview":
+                self.locales_list = self.request.event.settings.locales
+                for l in self.locales_list:
+                    self.output[l] = []
+                    self.output[l].append(form.cleaned_data['subject'].localize(l))
+                    message = form.cleaned_data['message'].localize(l)
+                    preview_text = (message.format(order='ORDER#', event='Your event name', order_date='2016-10-15', due_date='2016-10-31', order_url='link_to_order_url'))
+                    self.output[l].append(preview_text)
+                return self.get(self.request, *self.args, **self.kwargs)
```

Comment:
You could improve here even further by acually switching to that locale before you generate the text:

```
from pretix.base.i18n import language
...

for l in self.request.event.settings.locales:
    with language(l):
        # Generate preview
```

This has the advantage that you can use actual data for some placeholders and it will be displayed in the correct language, eg.:

```
from datetime import timedelta
..

preview_text = message.format(
    order='ORDER1234', 
    event=self.request.event.name,
    due_date=date_format(now() + timedelta(days=7), 'SHORT_DATE_FORMAT'),
    ...
)
```